### PR TITLE
fix: Update `<FocusTrap>` to handle single focusable child element

### PR DESCRIPTION
**Pull request by maximo-macchi-cb**
## What changed? Why?
This PR adds a fallback check if there's only one focusable element in a `<FocusTrap>`. If there's no second focusable element, the `<FocusTrap>` continues its focus logic and does not error out.

A note has been added to the docs to warn customers that at least one child element should be focusable. This is done because adding fallback checks for the first focusable element adds a bit more complexity. It's an edge case we shouldn't have to consider because a `<FocusTrap>` should always have at least one focusable element. 

### Root cause (required for bugfixes)
There's a constant called `secondElement` used to handle focus logic for our dropdown components. This constant assumes there are at least two focusable children elements based on how it's queried: `const secondElement = focusableElements[1]`. When there's only one focusable child element, `secondElement` is `undefined` and so any reference to an attribute of `secondElement` such as `const secondElementIsMenuItemOrOption = secondElement?.role === 'menuitem' || secondElement?.role === 'option'` errors out.

## ~UI changes~

| iOS Old        | iOS New        |
| -------------- | -------------- |
| old screenshot | new screenshot |

| Android Old    | Android New    |
| -------------- | -------------- |
| old screenshot | new screenshot |

| Web Old        | Web New        |
| -------------- | -------------- |
| old screenshot | new screenshot |

### How has it been tested?

- [ ] Unit tests
- [ ] Interaction tests
- [ ] Pseudo State tests
- [x] Manual - Web
- [ ] Manual - Android (Emulator / Device)
- [ ] Manual - iOS (Emulator / Device)

## Illustrations/Icons Checklist

Required if this PR changes files under `packages/illustrations/**` or `packages/icons/**`

- [ ] verified visreg changes with Terran (include link to visreg run/approval)
- [ ] all illustration/icons names have been reviewed by Dom and/or Terran

## Change management

type=routine 
risk=low 
impact=sev5 

automerge=false

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>1</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
<tr>
<td><code>CODEOWNERS</code></td>
<td>
✅ See below
</td>
<td></td>
</tr>
</table>

<h2>
✅
<code>CODEOWNERS</code>
</h2>
<table>
<tr>
<th>Code Owner</th>
<th>Status</th>
<th>Calculation</th>
</tr>
<tr>

<td>ui-systems-eng-team</td>

<td>
✅
<code>1/1</code>
</td>


<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
Additional CODEOWNERS Requirement
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Sum</strong></td>
<td>
<strong>
0
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>From <code>CODEOWNERS</code></td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/web/src/overlays/FocusTrap.tsx [hunk 1]
```diff
@@ -239,7 +239,7 @@ export const FocusTrap = memo(function FocusTrap({
           (event.key === 'ArrowDown' &&
             (activeElementIsMenuItemOrOption || secondElementIsMenuItemOrOption))
         ) {
-          secondElement.focus();
+          secondElement?.focus();
```
Comment (hcopp) [inline packages/web/src/overlays/FocusTrap.tsx:242]: Would it be feasible to add tests for FocusTrap for when there are 0, 1, or 2+ children?

Comment (maximo-macchi-cb) [reply packages/web/src/overlays/FocusTrap.tsx:242]: Good idea! I've rewritten the test file to test `<FocusTrap>` more directly rather than using `<Dropdown>` (which is now deprecated). 

I didn't include tests for 0 children. Handling that edge case requires making more changes to the `<FocusTrap>` component. While not a huge lift, I don't think it's an edge case we should be considering. The purpose of the component is to trap focus on focusable elements. The note I added to the docs in this PR calls this out as well. I could add a warning when in a dev environment if that's helpful.
